datawig/imputer.py

```diff
@@ -173,6 +173,7 @@ def fit(self,
             train_df: pd.DataFrame,
             test_df: pd.DataFrame = None,
             ctx: mx.context = mx.gpu() if gpu_device() else mx.cpu(),
+            num_gpus: int = 1,
```

Comment:
changing the api is something that we should really only do if there is no other way around it. this seems like it could be handled by the utils.gpu_device function, no?

Reply:
yeah, i'd guess that using the maximum number of gpus is totally fine (in fact that was the behaviour at some point when using mx.gpu(), no?)

Reply:
Sure, I can do that. The only thing is, for longer training jobs, a user might want to use only 4 gpus and keep the remaining 4 for another experiment (rather than using all 8 at once). This is something I frequently did while experimenting. Let me know what you think, otherwise I can just use max number of gpus.

`mx.gpu()` in the prior implementation just defaults to `mx.gpu(0)`

Reply:
ah right, utils.gpu_device returns just one, but when using mx.gpu() as context, then all gpus are used no? so i just implemented it wrongly in that gpu_device function, but with mx.gpu() all gpus would be used automatically, right? i think that behaviour makes most sense. 

personally i would prefer to not change the api - the experiment case case makes sense, but in those scenarios i'd probably either not use a gpu for the 'other' experiment or just use another machine. but if everybody agrees on this, let's change the api.

Reply:
Oh sorry should've been clearer, basically when using mx.gpu() it defaults to using mx.gpu(0). So when you don't pass in a specific gpu number, it will default to use gpu 0. See here for reference: https://mxnet.incubator.apache.org/tutorials/basic/ndarray.html

Reply:
ah i see, thanks for the clarification. 

i still would probably use all available gpus, but again, if there's a consensus amongst everybody let's change the api.

Reply:
Why would leaving the choice to the user require an api change? If we want anything different from the maximum number of gpus, we pass the desired context to the function, otherwise we use the default argument(?)

I suppose we can have a convenience function that automatically creates the context but that function would be imported explicitly. Since this is a rather advanced features I don't see see a problem with that.

Reply:
by changing the api i meant changing the function signature of one of the central functions of the imputer, `fit`. of course this can be done in a backward compatible manner, but it cannot be undone. 

we already have a convenience function that is used as default arg for all contexts (`utils.gpu_device`), the suggestion was to change that function to return a list of contexts, which would be fine.